### PR TITLE
chore(deps): update to looplab/fsm 1.0.0

### PR DESCRIPTION
**Pull request by guilhem-barthes**
## Description

Between v0.3.0 and v1.0.0 (more precisely in https://github.com/looplab/fsm/pull/82), the authors of fsm introduced a new first parameter of type`context.Context` in `Callback`, `Event` and other structures. This PR:

- Create a wrapper that discard the first argument and call our callbacks
-  Add a blank Context ([Background()](https://pkg.go.dev/context#Background)) to all the event we create

## How has this been tested?

CI

## Checklist

- [x] [changelog](../CHANGELOG.md) was updated with notable changes
- [ ] documentation was updated

### CODE REVIEW

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Review by ThibaultFy — COMMENTED** [review]
why is that a breaking change ?

**Comment by guilhem-barthes** [comment]
> why is that a breaking change ?

Good point, it's only internal to the orchestrator

File: lib/service/computetaskstate.go [hunk 1]
```diff
@@ -79,13 +80,19 @@ func newState(updater taskStateUpdater, task *asset.ComputeTask) *fsm.FSM {
...
-			"after_transitionDone":   updater.onDone,
-			"after_transitionFailed": updater.onFailure,
+			"enter_state":            wrapFsmCallbackContext(updater.onStateChange),
+			"after_transitionDone":   wrapFsmCallbackContext(updater.onDone),
+			"after_transitionFailed": wrapFsmCallbackContext(updater.onFailure),
 		},
 	)
 }
 
+// wrapFsmCallbackContext wrap our previous updater funciton with an empty fsm Context (became an argument in v1.0.0)
```
Comment (ThibaultFy) [inline lib/service/computetaskstate.go:90]:
```suggestion
// wrapFsmCallbackContext wrap our previous updater function with an empty fsm Context (became an argument in v1.0.0)
```

**Review by ThibaultFy — COMMENTED** [review]
Could you explain a bit what's happening on the PR in the summary of the PR ? Might help me to understand and to review it 😇

**Review by ThibaultFy — APPROVED** [review]

**Pull request merged**